### PR TITLE
Issue-32/add-golang-lint

### PR DESCRIPTION
**Pull request by AugustasV**
#### Summary
In this pull request I will integrate golangcilint to circleci pipeline to check code. Didn't fixed any linter errors.

#### Ticket Link

  Add golang lint https://github.com/mattermost/matterbuild/issues/32

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -6,6 +6,9 @@ jobs:
     steps:
       - checkout
       - run: make check-style
+      
+orbs:
+  tools: gotest/tools@0.0.13
```
Comment (hanzei) [inline .circleci/config.yml]: CI fails with:
```
# Error calling workflow: 'untagged-build'
# Cannot find a definition for job named test
# 
# -------
# Warning: This configuration was auto-generated to show you the message above.
# Don't rerun this job. Rerunning will have no effect.
false
```
Could you please double check the syntax for this file?

**Review by hanzei — CHANGES_REQUESTED** [review]

**Comment by AugustasV** [comment]
```
INFO [config_reader] Config search paths: [./ /root/project /root /] 
INFO [config_reader] Used config file .golangci.yml 
ERRO Running error: no such linter "bodyclose"    
INFO Memory: 1 samples, avg is 68.9MB, max is 68.9MB 
INFO Execution took 5.867077ms    
```
@hanzei Pretty strange error, I will look into that. But I think I integrate golangci-lint.

**Comment by hanzei** [comment]
@AugustasV I think you have to specify the version of `golanci-lint` for Ci

**Comment by AugustasV** [comment]
> @AugustasV I think you have to specify the version of `golanci-lint` for Ci

@hanzei Thanks for suggestion, added and now it works.

File: server/server.go [hunk 2]
```diff
@@ -94,8 +94,8 @@ func WriteEnrichedResponse(w http.ResponseWriter, title, resp, color, style stri
 	w.Write(GenerateEnrichedSlashResponse(title, resp, color, style))
 }
 
-func PostExtraMessages(responseURL string, payload []byte) error {
-	req, err := http.NewRequest("POST", responseURL, bytes.NewBuffer(payload))
+func PostExtraMessages(ResponseURL string, payload []byte) error {
+	req, err := http.NewRequest("POST", ResponseURL, bytes.NewBuffer(payload))
```
Comment (hanzei) [inline server/server.go]: Usually variables are lowercased. Why not in this case?

File: .golangci.yml [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+issues:
+  exclude-rules:
+    - path: server/manifest.go
+      linters:
+        - deadcode
+        - unused
+        - varcheck
+    - path: server/configuration.go
+      linters:
+        - unused
```
Comment (hanzei) [inline .golangci.yml]: These rules are specific for plugins. Are they needed here?

File: .circleci/config.yml [hunk 4]
```diff
@@ -1,12 +1,18 @@
...
     docker:
       - image: circleci/golang:1.13
     steps:
       - checkout
-      - run: make check-style
-
+      - tools/install-golangci-lint:
+          golangci-lint-version: 1.31.0
+      - tools/lint:
+          golangci-lint-version: 1.31.0
```
Comment (hanzei) [inline .circleci/config.yml]: I would like to use the Makefile here, as that is also what developers are using. Is this suitable?

File: .circleci/config.yml [hunk 5]
```diff
@@ -91,4 +98,4 @@ workflows:
             tags:
               only: /^v.*/
             branches:
-              ignore: /.*/
+              ignore: /.*/
```
Comment (hanzei) [inline .circleci/config.yml]: Nit: please revert that change

File: .circleci/config.yml [hunk 6]
```diff
@@ -55,12 +61,13 @@ jobs:
 workflows:
   version: 2
   untagged-build:
-    jobs:
-      - lint
+   jobs:
+      - tools/lint:
+          golangci-lint-version: 1.31.0
```
Comment (hanzei) [inline .circleci/config.yml]: See above: I would like to stick with the Makfile

**Review by hanzei — CHANGES_REQUESTED** [review]
Thanks for fixing the CI issues :+1: 

A couple of comments below

File: server/server.go (hunk 2 above)
Comment (AugustasV) [reply server/server.go]: this was golanglint suggestion, that's I renamed it. Should I revert that change?

File: .circleci/config.yml (hunk 4 above)
Comment (AugustasV) [reply .circleci/config.yml]: But that's why Orb was used, to install golangcilint. Checking out code, installing golangcilint and running it.

File: .golangci.yml (hunk 3 above)
Comment (AugustasV) [reply .golangci.yml]: Thanks for notice. Deleted.

File: server/server.go (hunk 2 above)
Comment (hanzei) [reply server/server.go]: Yes please

File: .circleci/config.yml (hunk 4 above)
Comment (hanzei) [reply .circleci/config.yml]: Can the orb be used to _only_ install golangci-lint and use `make check-style` for running it?

**Review by AugustasV — COMMENTED** [review]
Fixed everything you mentioned, just forget to change ResponseURL to ResponseUrl in server.go file. But will do that in next commit, it's not breaking change. I think it should work great right now.

**Review by hanzei — DISMISSED** [review]
LGTM. Thanks for this contribution :tada:

**Comment by hanzei** [comment]
After taking with @AugustasV I enabled some more linter and fixed the arising issues myself. You can find the changes in [`16a071a` (#51)](https://github.com/mattermost/matterbuild/pull/51/commits/16a071aca7bcc2d09106805e86f4652c4ab9d466).

File: .golangci.yml [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+  misspell:
+    locale: US
+
+linters:
+  disable-all: true
+  enable:
+    - bodyclose
+    - deadcode
+    # - errcheck
+    # - goconst
```
Comment (agnivade) [inline .golangci.yml]: Let's remove the lines if we are not using them. Rather file tickets if we plan to enable them later.

**Review by agnivade — APPROVED** [review]

**Comment by hanzei** [comment]
@jaydeland Could you please review this PR?

**Review by jaydeland — APPROVED** [review]
LGTM

**Pull request merged**